Catalog item registration (#3656)
SkylineCommunications/dataminer-docs: user-guide/Cloud_Platform/Catalog/Register_Catalog_Item.md

When you register a new Catalog item, it will only become available for the registered organization, and it will be marked as *private* on the Catalog to indicate this.

Registration can be done using the [Catalog API](#using-the-catalog-api).

> [!TIP]
> For a practical example, refer to the tutorials [Registering a new connector in the Catalog](xref:Tutorial_Register_Catalog_Item) and [Registering a new version of a connector in the Catalog](xref:Tutorial_Register_Catalog_Version).

## Using the Catalog API

With the Catalog API, you can both [register a Catalog item](#registering-a-catalog-item-with-the-api) and [register a new version of a Catalog item](#registering-a-new-version-with-the-api).

>[!important]  
The API calls are authenticated using [organization keys](xref:Managing_DCP_keys#organization-keys). The key must have the *Register catalog items permission* and needs to be added to the HTTP request in a header called **Ocp-Apim-Subscription-Key**.  
You need to have the "Owner" role in order to access/create Organization keys.

> A Catalog item is identified by a unique id (Guid) which you should provide yourself.

Reviewer: Same for "ID" actually 😊
```suggestion
A Catalog item is identified by a unique ID (GUID), which you should provide yourself.
```

> A Catalog item is identified by a unique id (Guid) which you should provide yourself.

Author: should be fixed in current latest version

### Registering a Catalog item with the API

The register API call allows you to create or update a Catalog item, see [Registering a new version with the API](#registering-a-new-version-with-the-api) on how to add a version after successfully registering an item.

#### URL

<https://api.dataminer.services/api/key-catalog/v1-0/catalog/register>

#### HTTP method

PUT

#### Body

The body of the request should be of type **multipart/form-data** and must contain:  

A key of type **File** and name **file**.  

> The value of this key should be a **zip** file containing 

Reviewer 2: ok

> - [required] a ["manifest.yml"](xref:Register_Catalog_Item#manifest-file) file.

Reviewer 2: yes looks ok

> - [optional] a readme.md file that contains the description of the Catalog item.

Author: ok, fixed

> - [optional] a readme.md file that contains the description of the Catalog item.

Reviewer 2: The README.md is not yet in code block ` `

> - [optional] a readme.md file that contains the description of the Catalog item.

Reviewer 2: ok

> - [optional] an "images" folder containing any image being referenced by the readme file.  

Author: ok fixed

> - [optional] an "images" folder containing any image being referenced by the readme file.  

Reviewer 2: The images is not yet in code block ` `

> - [optional] an "images" folder containing any image being referenced by the readme file.  

Author: ok, updated to ` `

> - [optional] an "images" folder containing any image being referenced by the readme file.  

Reviewer 2: ok

Supported image extensions are *.jpg*, *.jpeg*, *.png*, *.gif*, *.bmp*, *.tif*, *.tiff* and *.webp*

##### Manifest file

> This file will contain all necessary information to register a Catalog item with a version. This manifest file should be a valid .yml file and will contain "required" and "optional" attributes to add extra information to the Catalog item. Note that limitations may apply to certain attributes based on length or formatting.

Author: ok

> This file will contain all necessary information to register a Catalog item with a version. This manifest file should be a valid .yml file and will contain "required" and "optional" attributes to add extra information to the Catalog item. Note that limitations may apply to certain attributes based on length or formatting.

Reviewer 2: ok

#   - enhancedservicemodel: If the catalog item is a DataMiner XML connector designed as DataMiner enhanced service model.
#   - visio: If the catalog item is a Microsoft Visio design.
#   - solution: If the catalog item is a DataMiner Solution.
#   - testingsolution: If the catalog item is a DataMiner Solution designed for automated testing and validation.
#   - samplesolution: If the catalog item is a DataMiner Solution used for training and education.
#   - standardsolution: If the catalog item is a DataMiner Solution which is a out-of-the-box solution for specific use case or application.
#   - dashboard: If the catalog item is a DataMiner Dashboard.
#   - lowcodeapp: If the catalog item is a DataMiner low-code app.
#   - datatransformer: If the catalog item is a Data Transformer.
#   - dataquery: If the catalog item is a GQI data query.
#   - functiondefinition: If the catalog item is a DataMiner function definition.
#   - scriptedconnector: If the catalog item is a DataMiner scripted connector.
#   - bestpracticesanalyzer: If the catalog item is a DataMiner Best practices Analysis file.

type: '<fill in type here>'

# [Required] 
# The ID of the catalog item.

> # All registered versions for the same ID are shown together in the catalog.

Reviewer: Ideally, it should be capitalized everywhere, to make it clear that we're referring to an official DataMiner module.

> # All registered versions for the same ID are shown together in the catalog.

Reviewer 2: ok

> # Registering a Catalog item

Reviewer 3: Should we not just remove all references to the UI registration since we are going to remove that anyway?

Author: i wasn't aware it was going to go away :) . better to delete it in that case indeed

In order to make a new Catalog item or a new version of a Catalog item available in the [DataMiner Catalog](https://catalog.dataminer.services/), you need to register it with an organization.

> When you register a new Catalog item, it will only become available for the registered organization, and it will be marked as *private* in the UI to indicate this.

Reviewer 2: ```suggestion
When you register a new Catalog item, it will only become available for the registered organization, and it will be marked as *private* on the Catalog to indicate this.
```

1. Once all information has been filled in, click *Register*.

   This will conclude the registration process for the Catalog item and return the unique identifier for the item in question. This identifier will allow you to register versions for the Catalog item or update additional properties using the [Catalog API](#catalog-api).

> ## Catalog API

Reviewer: The tutorials are supposed to provide a practical example of how you can apply procedures in the user guide, but they are not intended to replace the procedures. So while a link to the tutorial here would be handy, we shouldn't replace the description here with it.

1. Once all information has been filled in, click *Register*.

   This will conclude the registration process for the Catalog item and return the unique identifier for the item in question. This identifier will allow you to register versions for the Catalog item or update additional properties using the [Catalog API](#using-the-catalog-api).

## Using the Catalog API

With the Catalog API, you can both [register a Catalog item](#registering-a-catalog-item-with-the-api) and [register a new version of a Catalog item](#registering-a-new-version-with-the-api).

>[!important]  

> The API calls are authenticated using [organization keys](xref:Managing_DCP_keys#organization-keys). The key must have the *Register catalog items permission* and needs to be added to the HTTP request in a header called **Ocp-Apim-Subscription-Key**.  

Reviewer 2: Missing '>', the Important doesn't compile properly right now
![image](https://github.com/user-attachments/assets/716162d5-5d67-4972-9ec2-6af863e537da)

Author: prolly got fixed in between, looks ok now?

Reviewer 2: ok

> You need to have the "Owner" role in order to access/create Organization keys.

Reviewer 2: No capitalization necessary I believe for 'organization'

> A Catalog item is identified by a unique id (Guid) which can be obtained by the [Catalog UI](#using-the-catalog-ui) or created by yourself.

Reviewer 2: Will need to be updated to not mention catalog UI

Author: think it was updated just before

#   Cannot contain newlines.
#   Cannot contain leading or trailing whitespace characters.
tags:
  - '<fill in tag here>'
```

### Registering a new version with the API

The register version API call allows you to create a new version for a Catalog item

#### URL

> <https://api.dataminer.services/api/key-catalog/v1-0/catalog/register{catalogId:guid}/version>

Author: o crap good catch, it should be after catalog btw. fixed

> <https://api.dataminer.services/api/key-catalog/v1-0/catalog/register{catalogId:guid}/version>

Reviewer 2: ok

> Route parameter "catalogId" is the id of the catalog item of which a new version is registered. Must be a valid Guid

Reviewer 2: ok

#### HTTP method

POST

#### Body

The body of the request should be of type **multipart/form-data** and must contain  

- a key of type **File** with name **file**.  

> The value is the catalog item installation file, supported types are a DataMiner Protocol package (.dmprotocol) or a DataMiner Application package (.dmapplication)

Reviewer 2: ok
